src/topics/TopicEditModal.js

```diff
@@ -0,0 +1,129 @@
+// @flow strict-local
+import React, { useState, useContext, useEffect } from 'react';
+import { Modal, View } from 'react-native';
+import type { Node } from 'react';
+import styles, { ThemeContext, BRAND_COLOR, createStyleSheet } from '../styles';
+import { updateMessage } from '../api';
+import type { Auth, GetText, Stream } from '../types';
+import { fetchSomeMessageIdForConversation } from '../message/fetchActions';
+import ZulipTextIntl from '../common/ZulipTextIntl';
+import ZulipTextButton from '../common/ZulipTextButton';
+import Input from '../common/Input';
+
+type Props = $ReadOnly<{|
+  topicModalProviderState: {
+    visible: boolean,
+    topic: string,
+    streamId: number,
+  },
+  auth: Auth,
+  zulipFeatureLevel: number,
+  streamsById: Map<number, Stream>,
+  _: GetText,
+  closeEditTopicModal: () => void,
+|}>;
+
+export default function TopicEditModal(props: Props): Node {
+  const { topicModalProviderState, closeEditTopicModal, auth, zulipFeatureLevel, streamsById, _ } =
+    props;
+
+  const { visible, topic, streamId } = topicModalProviderState;
+
+  const [topicName, onChangeTopicName] = useState();
+
+  useEffect(() => {
+    onChangeTopicName(topic);
+  }, [topic]);
+
+  const { backgroundColor } = useContext(ThemeContext);
+
+  const modalStyles = createStyleSheet({
+    wrapper: {
+      flex: 1,
+      justifyContent: 'center',
+      alignItems: 'center',
+    },
+    modal: {
+      justifyContent: 'flex-start',
+      backgroundColor,
+      padding: 15,
+      shadowOpacity: 0.5,
+      shadowColor: 'gray',
+      shadowOffset: {
+        height: 5,
+        width: 5,
+      },
+      shadowRadius: 5,
+      borderRadius: 5,
+      width: '90%',
+    },
+    buttonContainer: {
+      flexDirection: 'row',
+      justifyContent: 'flex-end',
+    },
+    titleText: {
+      fontSize: 18,
+      lineHeight: 21,
+      color: BRAND_COLOR,
+      marginBottom: 10,
+      fontWeight: 'bold',
+    },
+  });
+
+  const handleSubmit = async () => {
+    if (topicName === '') {
```

Comment:
This `topicName === ''` expression appears here and also where we pass the `disabled` prop to a `ZulipTextButton`. Let's pull it out into a `const validationErrors = …` directly in the body of `TopicEditModal`, like we do in `ComposeBox`.

Then, let's make it a bit more sophisticated: `topicName.trim() === ''` (or equivalently, `topicName.trim().length === 0`). When the server rejects '' as a topic, it also rejects ' ', etc., so we should too, to avoid sending a request we know the server will reject.

Hmm, looking again at `ComposeBox`, I remember that sometimes servers _do_ accept an empty topic: in particular, if the realm [doesn't require topics in stream messages](https://zulip.com/help/require-topics). So you could follow `ComposeBox` by adding a

```js
const mandatoryTopics = useSelector(state => getRealm(state).mandatoryTopics);
```

and using `mandatoryTopics` in the new `const validationErrors`. So, specifically, if `topicName.trim() === ''` _and_ `mandatoryTopics`, then say the input is invalid because topics are required and no topic was given.

Reply:
You could also have `validationErrors` include an item for "the topic is too long," though I see you've handled that already with the `maxLength` prop on the input component, which seems OK. I slightly prefer doing that here for consistency, but could go either way.

Reply:
I've implemented `validationErrors`. Please take a look at error message wording. 📝